mava/components/jax/building/parameter_client.py

```diff
@@ -199,13 +196,15 @@ def on_building_trainer_parameter_client(self, builder: SystemBuilder) -> None:
                     get_keys.append(f"policy_{net_type_key}-{net_key}")
                     get_keys.append(f"critic_{net_type_key}-{net_key}")
 
-        # Add the optimizers to the variable server.
-        # TODO (dries): Adjust this if using policy and critic optimizers.
-        # TODO (dries): Add this back if we want the optimizer_state to
-        # be store in the variable source. However some code might
-        # need to be moved around as the builder currently does not
-        # have access to the opt_states yet.
-        # params["optimizer_state"] = trainer.store.opt_states
+                if net_type_key == "networks":
```

Comment:
Is this for if we are using shared network parameters?

Reply:
No, this is because in the previous mava code, we always updating opt_states for net_type_key "networks" hence I enforeced it here as well. As far as I can see, this outer key is always "networks" so this could be redudant. I created this issue to investiaget it: https://github.com/instadeepai/Mava/issues/748

Reply:
Thank you for clarifying :)